friends/src/Components/FriendForm.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+
+class FriendForm extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			friendInfo: {
+				name: '',
+				age: '',
+				email: ''
+			}
+		};
+	}
+	handleChange = (e) => {
+		this.setState({
+			friendInfo: {
+				...this.state.friendInfo,
+				[e.target.name]: e.target.value
+			}
+		});
+	};
+	updateFriend = (e) => {
```

Comment:
As far as I can tell, this `updateFriend` function is never called anywhere in the application, but you do have the right idea with attempting to send the entire `friendInfo` object to your `put` call.